clients/cli/devices.c

```diff
@@ -2366,9 +2347,6 @@ do_devices_disconnect (NmCli *nmc, int argc, char **argv)
 	if (nmc->timeout > 0)
 		info->timeout_id = g_timeout_add_seconds (nmc->timeout, device_op_timeout_cb, info);
 
-	g_signal_connect (nmc->client, NM_CLIENT_DEVICE_REMOVED,
-	                  G_CALLBACK (device_removed_cb), info);
```

Comment:
I think this was added precisely because the device could disappear without switching a state to DISCONNECTED. Are you sure this cannot happen?